Source/WebCore/platform/graphics/cg/UTIRegistry.cpp

```diff
@@ -107,14 +107,38 @@ MemoryCompactRobinHoodHashSet<String>& additionalSupportedImageTypes()
     return additionalSupportedImageTypes;
 }
 
+struct ImageTypeMimeTypePair {
+    ASCIILiteral imageType;
+    ASCIILiteral mimeType;
+};
+
+static const HashMap<String, String, ASCIICaseInsensitiveHash>& additionalImageTypeMimeTypeMap()
+{
+    static NeverDestroyed<HashMap<String, String, ASCIICaseInsensitiveHash>> mimeTypesMap = [] {
+        HashMap<String, String, ASCIICaseInsensitiveHash> map;
+        static constexpr ImageTypeMimeTypePair additionalPairs[] = {
+            { "com.adobe.photoshop-image"_s, "application/x-photoshop"_s },
+        };
+        for (auto& [imageType, mimeType] : additionalPairs)
+            map.add(imageType, mimeType);
+        return map;
+    }();
+    return mimeTypesMap;
+}
+
 void setAdditionalSupportedImageTypes(const Vector<String>& imageTypes)
 {
     MIMETypeRegistry::additionalSupportedImageMIMETypes().clear();
     for (const auto& imageType : imageTypes) {
         additionalSupportedImageTypes().add(imageType);
+
         auto mimeType = MIMETypeForImageType(imageType);
         if (!mimeType.isEmpty())
             MIMETypeRegistry::additionalSupportedImageMIMETypes().add(mimeType);
+
+        auto mapEntry = additionalImageTypeMimeTypeMap().find<ASCIICaseInsensitiveStringViewHashTranslator>(imageType);
+        if (mapEntry != additionalImageTypeMimeTypeMap().end())
+            MIMETypeRegistry::additionalSupportedImageMIMETypes().add(mapEntry->value);
```

Comment:
This is changing the mapping from 1:1 (UTI -> MimeType) to 1:M relation. So I would suggest moving this new relationship to a new function in UTIUtilities.cpp named something like `RequiredMIMETypesFromUTI()`. It can return a `HashSet<String>`. And we can replace the call above `MIMETypeForImageType()` by a call to this new function.